src/query/src/datafusion.rs

```diff
@@ -339,31 +341,36 @@ impl DatafusionQueryEngine {
     #[tracing::instrument(skip_all)]
     fn optimize_physical_plan(
         &self,
-        ctx: &mut QueryEngineContext,
+        _ctx: &mut QueryEngineContext,
         plan: Arc<dyn ExecutionPlan>,
     ) -> Result<Arc<dyn ExecutionPlan>> {
         let _timer = metrics::OPTIMIZE_PHYSICAL_ELAPSED.start_timer();
 
-        let state = ctx.state();
-        let config = state.config_options();
+        // TODO(ruihang): `self.create_physical_plan()` already optimize the plan, check
```

Comment:
Can we remove the commented-out code? @waynexia

Reply:
no hurry, let me figure out the reason later